Intersect.Client.Core/Interface/Game/Character/CharacterWindow.cs

```diff
@@ -476,37 +436,46 @@ public void UpdateExtraBuffs(Guid itemId)
                 {
                     case ItemEffect.CooldownReduction:
                         CooldownAmount += effect.Percentage;
-                        mCooldownReduction?.SetText(Strings.Character.CooldownReduction.ToString(CooldownAmount));
-
                         break;
                     case ItemEffect.Lifesteal:
                         LifeStealAmount += effect.Percentage;
-                        mLifeSteal?.SetText(Strings.Character.Lifesteal.ToString(LifeStealAmount));
-
                         break;
                     case ItemEffect.Tenacity:
                         TenacityAmount += effect.Percentage;
-                        mTenacity?.SetText(Strings.Character.Tenacity.ToString(TenacityAmount));
-
                         break;
                     case ItemEffect.Luck:
                         LuckAmount += effect.Percentage;
-                        mLuck?.SetText(Strings.Character.Luck.ToString(LuckAmount));
-
                         break;
                     case ItemEffect.EXP:
                         ExtraExpAmount += effect.Percentage;
-                        mExtraExp?.SetText(Strings.Character.ExtraExp.ToString(ExtraExpAmount));
-
                         break;
                     case ItemEffect.Manasteal:
                         ManaStealAmount += effect.Percentage;
-                        mManaSteal?.SetText(Strings.Character.Manasteal.ToString(ManaStealAmount));
-
                         break;
                 }
             }
         }
+
+        //Update tooltip with all buffs
+        UpdateExtraBuffsTooltip();
+    }
+
+    /// <summary>
+    /// Updates the tooltip text for the extra buffs label with all current buff values
+    /// </summary>
+    private void UpdateExtraBuffsTooltip()
```

Comment:
This method continues to update the values, even when it is not visible, all other tooltip behaviors out there like descwindows only update when it opens, so it would be better this way too, unify much of the update code here and update only when the hover happens, check if the button has hover enter and out events too and let's reduce updates code to update only when it opens, the fewer things to update the better.

probably good, if not all, the get update extra effects method can come here, remove the update calls and update only when opening